integration-tests/chopsticks/src/tests/switchPallet/pauseSwitch.test.ts

```diff
@@ -0,0 +1,366 @@
+	const { checkEvents } = withExpect(expect)
+
+	const switchParameters = getAssetSwitchParameters()
+
+	await setStorage(peregrineContext, {
+		...PeregrineConfig.assignNativeTokensToAccounts([keysAlice.address], initialBalanceKILT),
+		...PeregrineConfig.createAndAssignRocs(keysCharlie.address, []),
+		...PeregrineConfig.setSafeXcmVersion4(),
+		...PeregrineConfig.setSudoKey(keysAlice.address),
+	})
+
+	await setStorage(peregrineContext, PeregrineConfig.setSwitchPair(switchParameters))
+
+	await setStorage(assethubContext, {
+		...AssetHubConfig.assignDotTokensToAccounts(
+			[keysAlice.address, PeregrineConfig.siblingSovereignAccount],
+			initialBalanceROC
+		),
+		...AssetHubConfig.createForeignAsset(keysCharlie.address, [
+			[PeregrineConfig.siblingSovereignAccount, switchParameters.sovereignSupply],
+		]),
+	})
+
+	// 1. send ROCs 2 Peregrine
+	const peregrineDestination = { V4: getSiblingLocationV4(PeregrineConfig.paraId) }
+	const beneficiary1 = getAccountLocationV4(hexAddress(keysAlice.address))
+	const rocAsset = { V4: [getRelayNativeAssetIdLocationV4(ROC.toString())] }
+
+	const signedTx1 = assethubContext.api.tx.polkadotXcm
+		.limitedReserveTransferAssets(peregrineDestination, beneficiary1, rocAsset, 0, 'Unlimited')
+		.signAsync(keysAlice)
+
+	const events1 = await sendTransaction(signedTx1)
+
+	await createBlock(assethubContext)
+
+	// Should still pass
+	await checkEvents(events1, 'xcmpQueue').toMatchSnapshot(
+		`sender AssetHub::xcmpQueue::[XcmpMessageSent] asset ${JSON.stringify(rocAsset)}`
+	)
+	await checkEvents(events1, 'polkadotXcm').toMatchSnapshot(
+		`sender AssetHub::polkadotXcm::[Attempted,FeesPaid,Sent] asset ${JSON.stringify(rocAsset)}`
+	)
+	await checkEvents(events1, { section: 'balances', method: 'Withdraw' }).toMatchSnapshot(
+		`sender AssetHub::balances::[Withdraw] asset ${JSON.stringify(rocAsset)}`
+	)
+
+	await createBlock(peregrineContext)
+	const aliceRocBalance = await getFreeRocPeregrine(keysAlice.address)
+	expect(aliceRocBalance).toBeGreaterThan(BigInt(0))
+
+	// 2. Pause switch pair
+	await peregrineContext.api.tx.sudo
+		.sudo(peregrineContext.api.tx.assetSwitchPool1.pauseSwitchPair())
+		.signAndSend(keysAlice)
+	await createBlock(peregrineContext)
+
+	// 3. switch KILTs
+	const balanceToTransfer = initialBalanceKILT / BigInt(2)
+
+	const beneficiary = getAccountLocationV4(hexAddress(keysAlice.address))
+
+	let section: string = ''
+	let errorName: string = ''
+
+	// This should fail.
+	await peregrineContext.api.tx.assetSwitchPool1
+		.switch(balanceToTransfer.toString(), beneficiary)
+		.signAndSend(keysAlice, ({ dispatchError }) => {
+			if (dispatchError) {
+				const decoded = peregrineContext.api.registry.findMetaError(dispatchError.asModule)
+				section = decoded.section
+				errorName = decoded.name
+			}
+		})
+
+	await createBlock(peregrineContext)
+
+	expect(section).toBe('assetSwitchPool1')
+	expect(errorName).toBe('SwitchPairNotEnabled')
+
+	await checkSwitchPalletInvariant(expect)
```

Comment:
Why would a runtime upgrade break it? 

Instead of introducing a new invariant function, I changed the old one. The provided delta represents the amount of trapped assets: 

[9fed626](https://github.com/KILTprotocol/kilt-node/pull/665/commits/9fed626174260c857d75adb5ee3a2951619764ea)

Let me know if you have envisioned that.

Reply:
I also added some checks for balance movement for incoming switch requests: 

[fedafee](https://github.com/KILTprotocol/kilt-node/pull/665/commits/fedafeedd68883707be68a1fa976d8c1ddf9df0d)